godtools/App/Features/ToolDetails/Presentation/ToolDetails/ToolDetailsViewModel.swift

```diff
@@ -0,0 +1,252 @@
+            languagesRepository: languagesRepository
+        )
+        
+        getToolTranslationsUseCase.getToolTranslations(determineToolTranslationsToDownload: determineToolTranslationsToDownload, downloadStarted: { [weak self] in
+            // download started, currently nothing will be shown while this downloads in the background. ~Levi
+        }, downloadFinished: { [weak self] (result: Result<ToolTranslations, GetToolTranslationsError>) in
+            DispatchQueue.main.async { [weak self] in
+                
+                let hidesLearnToShareToolButtonValue: Bool
+                
+                switch result {
+                case .success(let toolTranslations):
+                    
+                    if let primaryLanguageTranslationManifest = toolTranslations.languageTranslationManifests.first {
+                        
+                        hidesLearnToShareToolButtonValue = primaryLanguageTranslationManifest.manifest.tips.isEmpty
+                    }
+                    else {
+                        
+                        hidesLearnToShareToolButtonValue = true
+                    }
+                    
+                case .failure(let error):
+                    
+                    hidesLearnToShareToolButtonValue = true
+                }
+                
+                self?.hidesLearnToShareToolButton = hidesLearnToShareToolButtonValue
+            }
+        })
+    }
+    
+    private func reloadFavorited() {
+                
+        if favoritedResourcesCache.isFavorited(resourceId: resource.id) {
+            hidesAddToFavoritesButton = true
+            hidesRemoveFromFavoritesButton = false
+        }
+        else {
+            hidesAddToFavoritesButton = false
+            hidesRemoveFromFavoritesButton = true
+        }
+    }
+    
+    private func reloadToolDetails() {
+        
+        let resourcesCache: ResourcesCache = dataDownloader.resourcesCache
+
+        let nameValue: String
+        let aboutDetailsValue: String
+        let languageBundle: Bundle
+        
+        if let primaryLanguage = languageSettingsService.primaryLanguage.value, let primaryTranslation = resourcesCache.getResourceLanguageTranslation(resourceId: resource.id, languageId: primaryLanguage.id) {
+            
+            nameValue = primaryTranslation.translatedName
+            aboutDetailsValue = primaryTranslation.translatedDescription
+            languageBundle = localizationServices.bundleLoader.bundleForResource(resourceName: primaryLanguage.code) ?? Bundle.main
+        }
+        else if let englishTranslation = resourcesCache.getResourceLanguageTranslation(resourceId: resource.id, languageCode: "en") {
+            
+            nameValue = englishTranslation.translatedName
+            aboutDetailsValue = englishTranslation.translatedDescription
+            languageBundle = localizationServices.bundleLoader.englishBundle ?? Bundle.main
+        }
+        else {
```

Comment:
I see this pattern a lot:
```
if let primary language = ..., let primaryTranslation = ... {

} else if let englishTranslation = ... {

} else {
   (use resource itself)
}
```

I had been trying to figure out if there was a stop we could put this logic so we didn't have it in so many places, but I'm not sure where it would make sense.  I tried to do this for the language bundle logic in #1160, but maybe it should all go in some other class that holds helper logic like this?

Reply:
Hey @rachaelblue this looks like another place where we could use shared use case extensions.  I did add a getPrimaryLanguageBundle to this PR here (https://github.com/CruGlobal/godtools-swift/pull/1176).   I can create separate use cases for the translated name and description.

Reply:
Definitely don't want to be duplicating code.  I'll create a separate task for this and look into consolidating that.